lib/route/src/Obelisk/Route.hs

```diff
@@ -577,32 +580,56 @@ checkEnum1EncoderFunc f = do
 
 -- | This type is used by pathComponentEncoder to allow the user to indicate how to treat
 -- various cases when encoding a dependent sum of type `(R p)`.
-data SegmentResult check parse a =
-    PathEnd (Encoder check parse a (Map Text (Maybe Text)))
+data SegmentResult check parse a b =
```

Comment:
Breaking change - migration while keeping old meaning is to use `(Map Text (Maybe Text))` for `b`, but sometimes this can be generalized to also work for `()`, as with `indexOnlyRouteSegment`